src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -120,12 +147,64 @@ def _detect_performance_issue(data: Event, sdk_span: Any):
         DetectorType.N_PLUS_ONE_SPANS: NPlusOneSpanDetector(detection_settings, data),
     }
 
+    # Create performance issues for duplicate spans first
+    used_perf_issue_detectors = {
+        DetectorType.DUPLICATE_SPANS_HASH: DuplicateSpanHashDetector(detection_settings, data),
+        DetectorType.DUPLICATE_SPANS: DuplicateSpanDetector(detection_settings, data),
+    }
+
     for span in spans:
         for _, detector in detectors.items():
             detector.visit_span(span)
 
     report_metrics_for_detectors(event_id, detectors, sdk_span)
 
+    detected_problems = [
+        (i, detector_type)
+        for detector_type, d in used_perf_issue_detectors.items()
+        for i in d.stored_issues
+    ]
+
+    truncated_problems = detected_problems[:PERFORMANCE_GROUP_COUNT_LIMIT]
+
+    formatted_perf_problems = [
+        prepare_problem_for_grouping(problem, data) for problem, _ in truncated_problems
+    ]
+    return formatted_perf_problems
+
+
+def prepare_problem_for_grouping(problem: PerformanceSpanProblem, data: Event):
+    transaction_name = data.get("transaction")
+    first_span_id = problem[2][0]
```

Comment:
Problem spans can be accessed by attr instead of index, and we should imo, having unnamed index lookups is usually surprising, it's not really a perf benefit to use list / index lookup here.

Reply:
👍 addressed